### DISCUSSION TITLE
Libraries competing for I2S?

### DISCUSSION THREAD

**Discussion by renatoianhez** [schreibfaul1/ESP32-audioI2S#734] [Q&A]
With the new version of the ESP32-Arduino IDE, there was a change in the I2S API, and consequently some libraries were affected. One of them that I use is [pschatzmann's Bluetooth A2DP](https://github.com/pschatzmann/ESP32-A2DP). On [this version of my webradio](https://github.com/renatoianhez/Lolin32-radio/tree/main/RadioWebTestesESP3) I use it to transform it into a bluetooth receiver speaker. However, updating that library seems to conflict with your ESP32-audioI2S library, since, even in exclusive programming blocks, the Bluetooth library does not work. However, the A2DP works in an [example](https://github.com/pschatzmann/ESP32-A2DP/tree/main/examples/bt_music_receiver_simple_with_pin) sketch. Is there a "competition" for I2S in this case?

**Comment by schreibfaul1** [comment]
That may be, in V3 there is a handle on I2S. A double initialisation could lead to an error. You can create an audioI2S object as a reference.
````c++
Audio* audio = new(Audio);
audio->setPinout(1,2,3);
audio->connecttoFS(SD_MMC, "/mySong.mp3");
....

delete audio;
....

BT* bt = new(BT);
 ````
 You can destroy the object and thus release I2S. You can then create a Bluetooth object.

**Comment by renatoianhez** [comment]
Oh! Thank you! I will test something similar!
